kolibri/tasks/api.py

```diff
@@ -190,6 +183,18 @@ def localdrive(self, request):
 
         return Response(out)
 
+    @list_route(methods=['get'])
+    def channelinfo(self, request):
+        """
+        Gets metadata about a channel through a token or channel id.
+        """
+        url = get_channel_lookup_url(request.data['channel_id'])
+        resp = requests.get(url)
```

Comment:
Are you able to add caching here using Django's [cache](https://docs.djangoproject.com/en/1.11/topics/cache/) framework? I think it's a danger to make a network request that's synchronous with the currently ongoing request. In the short term, a hybrid works best: Make a request to Studio with a low timeout (0.2 seconds?) -- if it loads, great, return that and store that in the cache. If it timeouts, then return whatever's in the cache. Wait for a request no matter how long it takes, if we don't have anything stored in the cache yet.

What I'm thinking is to

Reply:
> danger to make a network request that's synchronous with the currently ongoing request
can you expand on this @aronasorman?
Maybe has to do with my unfamiliarity with the Tasks api and icecube, but we are doing a synchronous request within an asynchronous request?